### ISSUE TITLE
Pusher tries to reconnect after coming out of sleep on PC

### ISSUE THREAD

**Issue by johnmurphy01** [digitalcreations/Pusher.NET#4]
I am writing a WPF .NET application and have been using Pusher for the websocket implementation. I would call this issue an edge case but it's happening often enough that I really want to fix it. Whenever the PC goes to sleep and then returns out of sleep, the app crashes and it looks to be due to Pusher trying to reconnect the socket even though it thinks it is already connected. Here is the stack trace from the Event Viewer:
```
--FROM CLIENT-- INFO: System.Exception: The socket is connected, you neednt' connect again!
at Microsoft.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at Microsoft.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccess(Task task)
at Pusher.Pusher.<<.ctor>b__2>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.b__6_1(Object state)
at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
at System.Threading.ThreadPoolWorkQueue.Dispatch()
at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback() AT 6/29/2017 9:12:36 PM
```

`The socket is connected, you neednt' connect again!` I found in the SuperSocket source code which Pusher.NET uses. Any guidance here would be immensely helpful. I have everything wrapped in exception handling but the app still crashes and I'm not sure why. It also doesn't help that I cannot personally reproduce this issue.

**Comment by johnmurphy01** [comment]
I ended up having to create a method that pings the internet and if the app can't reach google.com, then I disconnect from Pusher. I would have assumed Pusher.NET had some way to handle this, but it doesn't look like it does.

**Comment by vegardlarsen** [comment]
@johnmurphy01 Thanks for letting me know of this issue. As you may be able to tell from this repo, this is no longer under active development. We made this library for a client's project, a project that has since been discontinued. We are therefore not maintaining it, but it should work.

If you do have the possibility to implement a proper fix for your issue, please let me know and I will attempt to create a release for it.

**Comment by johnmurphy01** [comment]
@vegardlarsen thanks for replying. As I mentioned, I had to implement a method that determines whether my app is connected to the internet. If it is not, I disconnect from pusher. Once the network is re-established, I re-connect to the websocket using Pusher.

Ideally, I'd love for Pusher.NET to handle this for me. But I can always fork the project if it bothers me too much. Thanks again!

**Comment by johnmurphy01** [comment]
Re-opened and created a PR(#5) to fix

**Comment by vegardlarsen** [comment]
I have merged your pull request and published version 0.3.2 with the changes to Nuget.

**Comment by johnmurphy01** [comment]
Thanks!
